test/e2e/etcdquorumguard_test.go

```diff
@@ -0,0 +1,279 @@
+)
+
+type podstatus struct {
+	node   string
+	status corev1.PodPhase
+}
+
+var nodes = make(map[string]bool)
+var pods = make(map[string]podstatus)
+
+type podinfo map[string]podstatus
+
+func TestEtcdQuorumGuard(t *testing.T) {
+	kclient, err := initialize()
+	if err != nil {
+		fmt.Printf("No etcd-quorum-guard deployment present; assume for now it is not configured.\n")
+		return
+		//t.Fatal(err.Error())
+	}
+	fmt.Print("Make all schedulable\n")
+	if err = makeAllNodesSchedulable(kclient); err != nil {
+		t.Errorf("Unable to make all nodes schedulable: %s", err.Error())
+	}
+	fmt.Print("Check for all running\n")
+	if err = waitForPods(kclient, 3, 3, 3); err != nil {
+		t.Errorf("Unable to get all etcd-quorum-guard pods running: %s", err.Error())
+	}
+	fmt.Print("Make one unschedulable\n")
+	if err = makeOneNodeUnschedulable(kclient); err != nil {
+		t.Errorf("Unable to make one node unschedulable: %s", err.Error())
+	}
+	fmt.Print("Wait for 2 running\n")
+	if err = waitForPods(kclient, 3, 2, 2); err != nil {
+		t.Errorf("Unable to get one etcd-quorum-guard pod stopped: %s", err.Error())
+	}
+	fmt.Print("Make second unschedulable\n")
+	if err = makeOneNodeUnschedulable(kclient); err == nil || !strings.Contains(err.Error(), "it would violate the pod's disruption budget") {
+		fmt.Print("  Pod should not have been evicted\n")
+		t.Errorf("Pod should not have been evicted because it violated disruption budget: %v", err)
+	} else {
+		fmt.Print("  Eviction correctly failed because it would violate the pod's disruption budget.\n")
+	}
+	fmt.Print("Make all schedulable\n")
+	if err = makeAllNodesSchedulable(kclient); err != nil {
+		t.Errorf("Unable to make all nodes schedulable: %s", err.Error())
+	}
+	fmt.Print("Wait for all running\n")
+	if err = waitForPods(kclient, 3, 3, 3); err != nil {
+		t.Errorf("Unable to get all etcd-quorum-guard pods running: %s", err.Error())
+	}
+	fmt.Print("Make one unschedulable\n")
+	if err = makeOneNodeUnschedulable(kclient); err != nil {
+		t.Errorf("Unable to make one node unschedulable: %s", err.Error())
+	}
+	fmt.Print("Wait for one not running\n")
+	if err = waitForPods(kclient, 3, 2, 2); err != nil {
+		t.Errorf("Unable to get one etcd-quorum-guard pod stopped: %s", err.Error())
+	}
+	fmt.Print("Make all schedulable\n")
+	if err = makeAllNodesSchedulable(kclient); err != nil {
+		t.Errorf("Unable to make all nodes schedulable: %s", err.Error())
+	}
+	fmt.Print("Wait for all\n")
+	if err = waitForPods(kclient, 3, 3, 3); err != nil {
+		t.Errorf("Unable to get all etcd-quorum-guard pods running: %s", err.Error())
+	}
+}
+
+func initialize() (*k8sclient.Clientset, error) {
+	kubeconfig := os.Getenv("KUBECONFIG")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("Error initializing kubeconfig: %s\n", err.Error())
+	}
+	kclient, err := k8sclient.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating client: %s\n", err.Error())
+	}
```

Comment:
we do have this initialization in e2e, you can reuse

Reply:
Yup.